index.js

```diff
@@ -68,7 +87,16 @@ function checkIfAnyMovieHasRating() {}
       // Toy Story 4
     };
  */
-function findById() {}
+function findById(movies, id) {
+  if(movies.length===0){
+    throw "Error Empty Array"
+  }
+  if(movies.imbID !== id){
+    return null
+  }
+  let findId = movies.find(movie=>movie.imbID === id)
+  return findId
```

Comment:
This part is good! But it's actually "movie.imdbID", not "movie.imbID". The only other thing you're missing is this:
```suggestion
  let findId = movies.find(movie=>movie.imdbID === id)
  if(findId === undefined) {
  return null
  } else {
  return findId
  }
```
If the .find method can't find a match, it returns undefined. So you have to check if findId is undefined. If it is, that means that no movie has an id that matches the input, so you need to return null. Otherwise just return findId.